### PR TITLE
Unit Testing Updates

### PR DESCRIPTION
**Pull request by Delphinus8821**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
Unit tests for CorePluginReporterView and CoreWelcomeView

### Why Should This Be In Core?
More unit tests in core

### Applicable Issues
#413

### CODE REVIEW

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/PluginReporterTopComponentNGTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        }
+    }
+
+    /**
+     * Test of handleComponentOpened method, of class PluginReporterTopComponent.
+     */
+    @Test
+    public void testHandleComponentOpened() {
+        System.out.println("handleComponentOpened");
+        PluginReporterTopComponent instance = mock(PluginReporterTopComponent.class);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/PluginReporterTopComponentNGTest.java:68]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/PluginReporterTopComponentNGTest.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        }
+    }
+
+    /**
+     * Test of componentShowing method, of class PluginReporterTopComponent.
+     */
+    @Test
+    public void testComponentShowing() {
+        System.out.println("componentShowing");
+        PluginReporterTopComponent instance = mock(PluginReporterTopComponent.class);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/PluginReporterTopComponentNGTest.java:92]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/PluginReporterTopComponentNGTest.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        }
+    }
+
+    /**
+     * Test of createStyle method, of class PluginReporterTopComponent.
+     */
+    @Test
+    public void testCreateStyle() {
+        System.out.println("createStyle");
+        PluginReporterTopComponent instance = mock(PluginReporterTopComponent.class);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/PluginReporterTopComponentNGTest.java:116]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        }
+    }
+
+    /**
+     * Test of toString method, of class PluginReportPane.
+     */
+    @Test
+    public void testToString() {
+        System.out.println("toString");
+        PluginReporterPane pluginReporterPane = mock(PluginReporterPane.class);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:61]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        PluginReport pluginReport = mock(PluginReport.class);
+        PluginReportPane instance = new PluginReportPane(pluginReporterPane, pluginReport, null, null);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:63]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    /**
+     * Test of toString method, of class PluginReportPane.
+     */
+    @Test
+    public void testToString() {
+        System.out.println("toString");
+        PluginReporterPane pluginReporterPane = mock(PluginReporterPane.class);
+        PluginReport pluginReport = mock(PluginReport.class);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:62]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        PluginReportPane instance = new PluginReportPane(pluginReporterPane, pluginReport, null, null);
+        
+        when(pluginReport.getPluginName()).thenReturn("plugin name");
+        String expResult = "plugin name";
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:66]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getPluginReport method, of class PluginReportPane.
+     */
+    @Test
+    public void testGetPluginReport() {
+        System.out.println("getPluginReport");
+        PluginReporterPane pluginReporterPane = mock(PluginReporterPane.class);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:77]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+        PluginReport pluginReport = mock(PluginReport.class);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:78]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+        PluginReportPane instance = new PluginReportPane(pluginReporterPane, pluginReport, null, null);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:79]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+        PluginReport result = instance.getPluginReport();
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:81]: Add final

File: CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+        assertEquals(result, pluginReport);
+    }
+
+    /**
+     * Test of updateTime method, of class PluginReportPane.
+     */
+    @Test
+    public void testUpdateTimeHour() {
+        System.out.println("updateTime");
+        PluginReporterPane pluginReporterPane = mock(PluginReporterPane.class);
```
Comment (sol695510) [inline CorePluginReporterView/test/unit/src/au/gov/asd/tac/constellation/views/pluginreporter/panes/PluginReportPaneNGTest.java:91]: Add final

**Review by sol695510 — APPROVED** [review]

**Review by OrionsGuardian — APPROVED** [review]

**Pull request merged**